inference-engine/tests/ie_test_utils/common_test_utils/ngraph_test_utils.cpp

```diff
@@ -83,6 +85,525 @@ std::string name(const Node& n) {
+               storage<std::vector<float>>().get_attributes_number() +
+               storage<std::vector<double>>().get_attributes_number() +
+               storage<std::vector<std::string>>().get_attributes_number() +
+               storage<SubGraphOpInputDescription>().get_attributes_number() +
+               storage<SubGraphOpOutputDescription>().get_attributes_number() +
+               storage<SpecialBodyPorts>().get_attributes_number();
+    }
+};
+
+}  // namespace storage
+
+class ReadAndStoreAttributes : public ngraph::AttributeVisitor, protected storage::Storage {
+public:
+    void on_adapter(const std::string& name, ngraph::ValueAccessor<void>& adapter) override {
+        if (auto inputs =
+                ngraph::as_type<ngraph::AttributeAdapter<SubGraphOpInputDescription>>(&adapter)) {
+            insert(name, inputs->get());
+        } else if (
+            auto outputs =
+                ngraph::as_type<ngraph::AttributeAdapter<SubGraphOpOutputDescription>>(&adapter)) {
+            insert(name, outputs->get());
+        } else if (
+            auto ports = ngraph::as_type<ngraph::AttributeAdapter<SpecialBodyPorts>>(&adapter)) {
+            insert(name, ports->get());
+        } else {
+            m_read_result += "store   attr [ ERR ]: " + name +
+                             " [drop `void` comparison which is '" + adapter.get_type_info().name +
+                             "']";
+        }
+    }
+
+    void on_adapter(const std::string& name, ngraph::ValueAccessor<void*>& adapter) override {
+        const auto beg = static_cast<unsigned char*>(adapter.get_ptr());
+        const auto end = beg + adapter.size();
+        insert(name, storage::MemoryChunk{storage::MemoryChunk::Data(beg, end)});
+    }
+
+#define ON_ADAPTER(TYPE)                                                                      \
+    void on_adapter(const std::string& name, ngraph::ValueAccessor<TYPE>& adapter) override { \
+        insert(name, adapter.get());                                                          \
+    }
+
+    ON_ADAPTER(bool)
+    ON_ADAPTER(std::string)
+    ON_ADAPTER(int8_t)
+    ON_ADAPTER(int16_t)
+    ON_ADAPTER(int32_t)
+    ON_ADAPTER(int64_t)
+    ON_ADAPTER(uint8_t)
+    ON_ADAPTER(uint16_t)
+    ON_ADAPTER(uint32_t)
+    ON_ADAPTER(uint64_t)
+    ON_ADAPTER(float)
+    ON_ADAPTER(double)
+    ON_ADAPTER(std::vector<int8_t>)
+    ON_ADAPTER(std::vector<int16_t>)
+    ON_ADAPTER(std::vector<int32_t>)
+    ON_ADAPTER(std::vector<int64_t>)
+    ON_ADAPTER(std::vector<uint8_t>)
+    ON_ADAPTER(std::vector<uint16_t>)
+    ON_ADAPTER(std::vector<uint32_t>)
+    ON_ADAPTER(std::vector<uint64_t>)
+    ON_ADAPTER(std::vector<float>)
+    ON_ADAPTER(std::vector<double>)
+    ON_ADAPTER(std::vector<std::string>)
+
+#undef ON_ADAPTER
+
+    void on_adapter(
+        const std::string&, ngraph::ValueAccessor<std::shared_ptr<ngraph::Function>>&) override {
+        // handled by `compare_functions` drop it here
```

Comment:
I think it's better to compare nested functions as other attributes instead of checking it manually. I think it can be implemented in a separate PR.

Reply:
Is not enough to check subfunctions in https://github.com/openvinotoolkit/openvino/blob/a8b921791e4434cb2d092312c2d68a65004db85e/inference-engine/tests/ie_test_utils/common_test_utils/ngraph_test_utils.cpp#L157
?

Reply:
We store TI and Loop body as an attribute. So I suggest to compare it as a regular attribute.

Reply:
I guess it doesn't block us at all now.
Let's add this point to SubGraphOp comparison support ticket (@pelszkow if we haven't a such, please create)

Reply:
Will be covered by 47847.